crates/bevy_solari/src/pathtracer/pathtracer.wgsl

```diff
@@ -37,30 +37,42 @@ fn pathtrace(@builtin(global_invocation_id) global_id: vec3<u32>) {
     // Path trace
     var radiance = vec3(0.0);
     var throughput = vec3(1.0);
+    var p_bounce = 0.0;
+    var is_perfectly_specular = true;
+    var previous_normal = vec3(0.0);
     loop {
         let ray_hit = trace_ray(ray_origin, ray_direction, ray_t_min, RAY_T_MAX, RAY_FLAG_NONE);
         if ray_hit.kind != RAY_QUERY_INTERSECTION_NONE {
             let ray_hit = resolve_ray_hit_full(ray_hit);
             let wo = -ray_direction;
 
-            // Use emissive only on the first ray (coming from the camera)
-            if ray_t_min == 0.0 { radiance = ray_hit.material.emissive; }
+            var mis_weight = 1.0;
+            if !is_perfectly_specular {
+                let p_light = random_light_pdf(ray_hit);
+                mis_weight = power_heuristic(p_bounce, p_light);
+            }
+            radiance += mis_weight * throughput * ray_hit.material.emissive;
 
             // Sample direct lighting
             let direct_lighting = sample_random_light(ray_hit.world_position, ray_hit.world_normal, &rng);
+            let p_would_bounce = brdf_pdf(wo, direct_lighting.wi, ray_hit);
```

Comment:
```suggestion
            let pdf_would_bounce = brdf_pdf(wo, direct_lighting.wi, ray_hit);
```

So this is the probability that `wi` could've been sampled by a BRDF importance sample, right?

Reply:
Yup.